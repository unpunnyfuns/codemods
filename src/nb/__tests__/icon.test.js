import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/icon', () => {
  it('migrates Icon component', () => {
    const output = testTransform('nb/icon', 'nb/icon', 'tsx', {
      sourceImport: DEFAULT_TEST_OPTIONS.sourceImport,
      targetImport: '@target/components/Icon',
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
