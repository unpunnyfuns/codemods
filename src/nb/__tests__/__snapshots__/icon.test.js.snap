// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/icon > migrates Icon component 1`] = `
"import { Icon } from '@target/components/Icon';

export function Example() {
  return (
    <>
      {/* Basic usage */}
      <Icon name="CheckCircleSolid" width={5} height={5} color="icon.primary" />
      {/* With token sizes */}
      <Icon name="XCircleSolid" width={16} height={16} color="icon.secondary" />
      {/* With testID */}
      <Icon name="ArrowRight" width={20} height={20} color="text.primary" testID="arrow-icon" />
      {/* Color mapping */}
      <Icon name="Star" width={5} height={5} color="blue.500" />
      {/* With spread (will warn) */}
      <Icon name="Heart" width={5} height={5} color="red.500" />
    </>
  );
}
"
`;
